File: AndroidBase/androidBase/src/main/java/ams/android_base/utils/RoomUtils.java
package ams.android_base.utils;


import androidx.sqlite.db.SimpleSQLiteQuery;

@SuppressWarnings("unused")
public class RoomUtils {

    @SuppressWarnings("WeakerAccess")
    public static SimpleSQLiteQuery getQuery(String selectStatement) { return new SimpleSQLiteQuery(selectStatement); }

    public static SimpleSQLiteQuery selectAllQuery(String tableName) { return getQuery("SELECT * FROM" + tableName); }

    public static SimpleSQLiteQuery selectPageItemsQuery(String tableName, int itemsPerPage, int offset) {
        return getQuery("SELECT * FROM" + tableName + " LIMIT " + itemsPerPage + " OFFSET " + offset);
    }

    public static SimpleSQLiteQuery totalNumberOfRowsQuery(String tableName) {
        return getQuery("SELECT COUNT(*) FROM " + tableName);
    }

    public static SimpleSQLiteQuery totalNumberOfRowsQuery(String tableName, String condition) {
        return getQuery("SELECT COUNT(*) FROM " + tableName + condition);
    }

    public static SimpleSQLiteQuery deleteAllRowsQuery(String tableName) {
        return getQuery("DELETE FROM " + tableName);
    }

}

File: androiddevkit/src/main/java/com/ams/androiddevkit/utils/RoomUtils.kt
package com.ams.androiddevkit.utils

import androidx.sqlite.db.SimpleSQLiteQuery

open class RoomUtils {

    open fun getQuery(selectStatement: String?): SimpleSQLiteQuery {
        return SimpleSQLiteQuery(selectStatement)
    }

    open fun selectAllQuery(tableName: String): SimpleSQLiteQuery {
        return getQuery("SELECT * FROM$tableName")
    }

    open fun selectPageItemsQuery(tableName: String, itemsPerPage: Int, offset: Int): SimpleSQLiteQuery {
        return getQuery("SELECT * FROM$tableName LIMIT $itemsPerPage OFFSET $offset")
    }

    open fun totalNumberOfRowsQuery(tableName: String): SimpleSQLiteQuery {
        return getQuery("SELECT COUNT(*) FROM $tableName")
    }

    open fun totalNumberOfRowsQuery(tableName: String, condition: String): SimpleSQLiteQuery {
        return getQuery("SELECT COUNT(*) FROM $tableName$condition")
    }

    open fun deleteAllRowsQuery(tableName: String): SimpleSQLiteQuery {
        return getQuery("DELETE FROM $tableName")
    }
}